internal/v4a/middleware.go

```diff
@@ -71,10 +71,23 @@ func (s *SignHTTPRequestMiddleware) HandleFinalize(
 		return out, metadata, &SigningError{Err: fmt.Errorf("failed to retrieve credentials: %w", err)}
 	}
 
-	err = s.signer.SignHTTP(ctx, credentials, req.Request, payloadHash, signingName, []string{signingRegion}, time.Now().UTC(), func(o *SignerOptions) {
-		o.Logger = middleware.GetLogger(ctx)
-		o.LogSigning = s.logSigning
-	})
+	signerOptions := []func(o *SignerOptions){
+		func(o *SignerOptions) {
+			o.Logger = middleware.GetLogger(ctx)
+			o.LogSigning = s.logSigning
+		},
+	}
+
+	// existing DisableURIPathEscaping is equivalent in purpose
+	// to authentication scheme property DisableDoubleEncoding
+	disableDoubleEncoding, overridden := internalauth.GetDisableDoubleEncoding(ctx)
```

Comment:
**correctness**: This change to allow overriding this setting should apply to v4 as well should it not?

Reply:
agh ... oversight. fixing.